### PR TITLE
Add API endpoint to get latest release

### PR DESCRIPTION
**Pull request by JakobDev**
This PR adds a new API endpoint to get the latest stable release of a repo, similar like the [GitHub API](https://docs.github.com/en/rest/releases/releases#get-the-latest-release).

### CODE REVIEW

**Review by delvh — COMMENTED** [review]
1. A test would be nice
2. Please run `make fmt`

Then I'll approve.

**Comment by JakobDev** [comment]
@delvh In which directory are the tests for the API?

**Comment by delvh** [comment]
`tests/integration/api_<feature>_test.go`

**Comment by JakobDev** [comment]
I have added tests now.

`make fmt `didn't change anything.

**Review by lafriks — APPROVED** [review]

File: tests/integration/release_test.go [hunk 1]
```diff
@@ -81,6 +81,21 @@ func TestViewReleasesNoLogin(t *testing.T) {
...
 
+func TestViewLatestRelease(t *testing.T) {
+	defer tests.PrepareTestEnv(t)()
+
+	session := loginUser(t, "user2")
+	req := NewRequest(t, "GET", "/user2/repo1/releases/latest")
+	session.MakeRequest(t, req, http.StatusOK)
+}
+
+func TestViewLatestReleaseNoLogin(t *testing.T) {
```
Comment (delvh) [inline tests/integration/release_test.go]: I can see one to two more test cases:
1. non-public releases are omitted when not logged in. Potentially overkill.
2. draft releases are not returned (even for site admins/ users with write rights).

File: routers/api/v1/repo/release.go [hunk 2]
```diff
@@ -68,6 +68,47 @@ func GetRelease(ctx *context.APIContext) {
 	ctx.JSON(http.StatusOK, convert.ToRelease(release))
 }
 
+//  GetLatestRelease gets the latest stable release of a repository
```
Comment (delvh) [inline routers/api/v1/repo/release.go]: I think here is the problem the CI has:
```suggestion
// GetLatestRelease gets the latest stable release of a repository
```

File: routers/api/v1/repo/release.go [hunk 3]
```diff
@@ -68,6 +68,47 @@ func GetRelease(ctx *context.APIContext) {
 	ctx.JSON(http.StatusOK, convert.ToRelease(release))
 }
 
+// GetLatestRelease gets the latest stable release of a repository
```
Comment (jolheiser) [inline routers/api/v1/repo/release.go]:
```suggestion
// GetLatestRelease gets the most recent non-prerelease, non-draft release of a repository, sorted by created_at
```
This is similar to how GitHub words it, and I think it should make it less ambiguous for users

File: routers/api/v1/repo/release.go [hunk 4]
```diff
@@ -68,6 +68,47 @@ func GetRelease(ctx *context.APIContext) {
...
+func GetLatestRelease(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/releases/latest repository repoGetLatestRelease
+	// ---
+	// summary: Gets the latest stable release
```
Comment (jolheiser) [inline routers/api/v1/repo/release.go]:
```suggestion
	// summary: Gets the most recent non-prerelease, non-draft release of a repository, sorted by created_at
```

**Review by jolheiser — COMMENTED** [review]
This is something I was _just_ looking for, and this PR seems nearly ready to go. :rocket: 
Thanks for the work!

**Comment by JakobDev** [comment]
Sorry for the long silent.

I had now applied the comments from @jolheiser.

To write the tests from @delvh , I would need time to dig into the test suite, so I hope this is not a blocker.

**Review by jolheiser — APPROVED** [review]
Assuming CI passes I think this is fine to go.

**Comment by jolheiser** [comment]
Ah, didn't notice I would be the second approver, I would like @delvh opinion since he reviewed earlier. 
@delvh do you have further reservations on the test(s)?

**Comment by jolheiser** [comment]
🎺 🤖

**Pull request merged**